project/src/main/java/com/google/autograder/servlets/auth/AccessTokenResponse.java

```diff
@@ -55,13 +57,23 @@ public static Entity createDatastoreAccessTokenResponseEntity(AccessTokenRespons
         accessTokenResponseEntity.setProperty("expires_in", accessTokenResponse.expires_in);
         accessTokenResponseEntity.setProperty("scope", accessTokenResponse.scope);
         accessTokenResponseEntity.setProperty("user_email", userEmail);
+
+        Instant expires = Instant.now().plus(accessTokenResponse.expires_in, ChronoUnit.SECONDS);
+        accessTokenResponseEntity.setProperty("expires", expires.toString());
+        
         return accessTokenResponseEntity;
     }
     
     public static AccessTokenResponse buildAccessTokenResponseFromDatastoreEntity(Entity entity) {
         if (entity.getProperty("access_token") == null) {
             return null;
         }
+
+        Instant expires = Instant.parse((String) entity.getProperty("expires"));
+
+        if (expires.getEpochSecond() - Instant.now().getEpochSecond() <= 0) {
```

Comment:
You can do something like `expires.isAfter(Instant.now())`. One of the main points of using time libraries and other specialized classes is to avoid having to perform arithmetic on primitives, which is more error prone

Reply:
+1